packages/sidebar/src/Sidebar.tsx

```diff
@@ -112,15 +130,16 @@ export const PureSidebar: React.FC<ISidebarProps> = ({
 
   const handleClose = useCallback(() => {
     setAnimate(true);
-
     if (arrowToRight) {
-      setWidth(maxWidth);
-      setArrowToRight(false);
+      setWidth(propsWidth);
     } else {
-      setWidth(minWidth);
-      setArrowToRight(true);
+      setWidth(rollWidth);
     }
-  }, [arrowToRight, maxWidth, minWidth]);
+    onChangeWidth({
+      collapsed: !arrowToRight,
+      width: widthRef.current < minWidth ? minWidth + 1 : widthRef.current,
```

Comment:
Logically, there must be width: arrowToRight ? propsWidth : rollWidth.

Reply:
only  `width: propsWidth,` :)